app/controllers/course/assessment/assessments_controller.rb

```diff
@@ -151,4 +152,39 @@ def question_assessments_hash
   def valid_ordering?(proposed_ordering)
     question_assessments_hash.keys.sort == proposed_ordering.sort
   end
+
+  # Mapping of `tab_id`s to their compound titles. If the tab is the only one in its category,
+  # the category title is used. Otherwise, the category is prepended to the tab title.
+  #
+  # @return [Hash{Integer => String}]
+  def compound_tab_titles
```

Comment:
Would it be better to place these functions into helper functions as they are used more for displays?

Reply:
I think it is better placed in the controller because it does DB fetching. Also, the ordering step of `ordered_assessments_by_tab` depends on this method, so if we want to make this a view helper method in the typical sense (i.e. called form the view but preferably not needed in the controller), I'll have to rework the method. Would it help if I just wrapped `tabs.length > 1 ? "#{category_title} - #{title}" : category_title` in a helper method?